### PR TITLE
change return value of Type::pointee_type() to Option<Type>

### LINKED ISSUE #137: clang::Type::pointee_type should return Option<Type>
Right now it always does an FFI call to `clang_getPointeeType` but this is invalid if this type is not a pointer type. We should check that, and if it isn't return `None`, and if it is make the FFI call and return `Some`.

I can mentor whoever is interested in working on this.

### PR DESCRIPTION
**Pull request by ajnirp**
Fixes #137

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
Please claim the relevant issue before working on it, so that we can reduce duplicate efforts :)

File: src/clang.rs [hunk 1]
```diff
@@ -647,11 +647,17 @@ impl Type {
 
     /// Given that this type is a pointer type, return the type that it points
     /// to.
-    pub fn pointee_type(&self) -> Type {
-        unsafe {
-            Type {
-                x: clang_getPointeeType(self.x),
+    pub fn pointee_type(&self) -> Option<Type> {
+        match self.x.kind {
```
Comment (emilio) [inline src/clang.rs]: can you use `self.kind()`? not a big deal in this case, but better to be consistent.

File: src/ir/ty.rs [hunk 2]
```diff
@@ -677,21 +677,19 @@ impl Type {
...
             // can even add bindings for that, so huh.
             CXType_RValueReference |
             CXType_LValueReference => {
-                let inner = Item::from_ty_or_ref(ty.pointee_type(),
-                                                 location,
-                                                 parent_id,
-                                                 ctx);
+                let inner =
+                    Item::from_ty_or_ref(ty.pointee_type().unwrap(), location,
+                                         parent_id, ctx);
```
Comment (Wafflespeanut) [inline src/ir/ty.rs]: Can't both the arms be merged?

**Comment by Wafflespeanut** [comment]
Never mind. Ignore my nits :)

File: src/clang.rs [hunk 3]
```diff
@@ -647,11 +647,17 @@ impl Type {
...
+            CXType_Pointer |
+            CXType_RValueReference |
+            CXType_LValueReference |
+            CXType_MemberPointer => {
+                unsafe {
+                    Some(Type { x: clang_getPointeeType(self.x) })
+                }
```
Comment (emilio) [inline src/clang.rs]: Huh, for some reason my other comment was lost.

Can you assert the type is valid here?

```rust
let ret = Type {
    x: unsafe { clang_getPointeeType(self.x) },
};
debug_assert!(ret.kind() != CXType_Invalid);
Some(ret)

Comment (ajnirp) [inline src/clang.rs]: Will do

File: src/clang.rs (hunk 1 above)
Comment (ajnirp) [inline src/clang.rs]: I didn't realise we had that method, thank you!

**Comment by emilio** [comment]
@bors-servo r+

Thanks!

**Pull request merged**